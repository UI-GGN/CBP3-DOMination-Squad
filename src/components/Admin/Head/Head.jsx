import close from '../../../assets/close.png';
import user from '../../../assets/user.png';
import {
  Container,
  AccessContainer,
  CrossContainer,
  NavigationContainer,
  Sections,
  ProfileContainer,
  ImageContainer,
  CrossImage,
  AccessTitle,
  ProfileImage,
  HorizontalLine,
  NavHeaderTitle,
  NavBar,
  NavBarOption,
  LogoutButton,
} from './Head.style';
import { useNavigate } from 'react-router-dom';

const Header = ({ selected, onPress, onHeaderClose }) => {
  const navigate = useNavigate();

  return (
    <Container data-testid="main">
      <AccessContainer>
        <CrossContainer>
          <CrossImage src={close} onClick={() => onHeaderClose()} data-testid="header-close" />
        </CrossContainer>
        <AccessTitle>Admin Access</AccessTitle>
      </AccessContainer>
      <HorizontalLine />
      <NavigationContainer>
        <Sections>
          <ProfileContainer>
            <ImageContainer>
              <ProfileImage src={user} />
            </ImageContainer>
            <NavHeaderTitle data-testid="profile-title">Hi Vinod!</NavHeaderTitle>
          </ProfileContainer>
          <HorizontalLine />
          <NavBar>
            <NavBarOption
              isSelected={selected === 'REQUESTS'}
              onClick={() => {
                onPress('REQUESTS');
              }}
              data-testid="header-nav-option-requests"
            >
              Requests
            </NavBarOption>
            <HorizontalLine />
            {/* <NavBarOption
              isSelected={selected === 'ROUTES'}
              onClick={() => {
                onPress('ROUTES');
              }}
              data-testid="header-nav-option-routes"
            >
              Routes
            </NavBarOption>
          <HorizontalLine /> */}
          </NavBar>
        </Sections>
        <LogoutButton
          onClick={() => {
            navigate('/login');
          }}
          data-testid="header-logout-button"
        >
          Log out
        </LogoutButton>
      </NavigationContainer>
    </Container>
  );
};

export default Header;
