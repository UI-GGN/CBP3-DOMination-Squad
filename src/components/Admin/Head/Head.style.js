import { primary, secondary, dark } from '../../colors.json';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 20%;
  background-color: ${dark};
  justify-content: space-between;
  padding-bottom: 8px;
`;

export const NavigationContainer = styled.div`
  background-color: ${dark};
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const AccessContainer = styled.div`
  background-color: #008080;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px;
  height: 8vh;
  overflow: auto;
`;

export const HorizontalLine = styled.div`
  height: 0.5px;
  background-color: ${secondary};
`;

export const AccessTitle = styled.h1`
  font-size: 1.6em;
  color: ${secondary};
  font-family: 'roboto-regular';
`;

export const CrossContainer = styled.div`
  height: 1em;
  align-self: end;
  margin-right: 4px;
  margin-top: 4px;
`;

export const CrossImage = styled.img`
  max-height: 80%;
  cursor: pointer;
`;

export const ProfileImage = styled.img`
  max-width: 60%;
`;

export const ProfileContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin: 8px 12px;
  align-items: center;
`;

export const ImageContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;
  width: 40px;
  border: 3px solid ${primary};
  border-radius: 20px;
`;

export const NavigationHeader = styled.div`
  display: flex;
  background-color: #1b3a4b;
`;

export const NavHeaderTitle = styled.h1`
  font-size: 20px;
  letter-spacing: 3px;
  margin: 12px;
  color: white;
`;

export const NavBar = styled.div`
  display: flex;
  flex-direction: column;
`;

export const NavBarOption = styled.button`
  font-size: 20px;
  font-family: 'roboto-regular';
  padding: 16px 8px;
  border-left: ${(props) => (props.isSelected ? `5px solid ${primary}` : '')};
  background-color: ${(props) => (props.isSelected ? secondary : '')};
  color: ${(props) => (props.isSelected ? dark : secondary)};
  &:hover {
    font-weight: ${(props) => (props.isSelected ? '' : '600')};
  }
`;

export const LogoutButton = styled.button`
  background-color: ${secondary};
  font-size: 20px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  padding: 6px;
  margin: 16px 0px;
  color: black;
  &:hover {
    background-color: #b23b3b;
    box-shadow: 0 10px 16px 0 rgba(0, 0, 0, 0.5);
    color: white;
  }
`;
