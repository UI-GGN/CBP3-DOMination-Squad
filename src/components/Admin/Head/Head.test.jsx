import Head from './Head';
import { render, fireEvent } from '@testing-library/react';
import { describe, it, vi, expect } from 'vitest';

const mocks = vi.hoisted(() => {
  return {
    useNavigate: vi.fn(),
  };
});

vi.mock('react-router-dom', async (importOriginal) => {
  const mod = await importOriginal();
  return {
    ...mod,
    useNavigate: mocks.useNavigate,
  };
});

describe('Head', () => {
  it('should render Head component', () => {
    const { container, getByTestId } = render(<Head />);

    expect(getByTestId('main')).toBeInTheDocument();
    expect(getByTestId('header-close')).toBeInTheDocument();
    expect(getByTestId('profile-title')).toBeInTheDocument();
    expect(getByTestId('header-nav-option-requests')).toBeInTheDocument();
    expect(getByTestId('header-logout-button')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should call onHeaderClose when close button is clicked', () => {
    const mockOnHeaderClose = vi.fn();
    const { getByTestId } = render(<Head onHeaderClose={mockOnHeaderClose} />);

    fireEvent.click(getByTestId('header-close'));

    expect(mockOnHeaderClose).toHaveBeenCalledTimes(1);
  });

  it('should call onPress when requests option is clicked', () => {
    const mockOnPress = vi.fn();
    const { getByTestId } = render(<Head selected="REQUESTS" onPress={mockOnPress} />);

    fireEvent.click(getByTestId('header-nav-option-requests'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(mockOnPress).toHaveBeenCalledWith('REQUESTS');
  });
});
