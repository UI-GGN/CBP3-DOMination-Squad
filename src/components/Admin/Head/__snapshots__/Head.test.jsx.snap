// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Head > should render Head component 1`] = `
<div>
  <div
    class="sc-beqWaB jwXsjE"
    data-testid="main"
  >
    <div
      class="sc-gueYoa jDqMaZ"
    >
      <div
        class="sc-gLDzan hRFYXL"
      >
        <img
          class="sc-iAEyYk hWbHTT"
          data-testid="header-close"
          src="/src/assets/close.png"
        />
      </div>
      <h1
        class="sc-jTrPJq eMSvBq"
      >
        Admin Access
      </h1>
    </div>
    <div
      class="sc-eDDNvR kfivcA"
    />
    <div
      class="sc-dmqHEX jFKZMM"
    >
      <div
        class="sc-hLseeU jmgCdO"
      >
        <div
          class="sc-qRumB gdVmCe"
        >
          <div
            class="sc-jsMahE lzUdM"
          >
            <img
              class="sc-fsQiph hVDqoN"
              src="/src/assets/user.png"
            />
          </div>
          <h1
            class="sc-irTswW btXJHY"
            data-testid="profile-title"
          >
            Hi Vinod!
          </h1>
        </div>
        <div
          class="sc-eDDNvR kfivcA"
        />
        <div
          class="sc-csCMJt elNvL"
        >
          <button
            class="sc-fbJfA hGHcJu"
            data-testid="header-nav-option-requests"
          >
            Requests
          </button>
          <div
            class="sc-eDDNvR kfivcA"
          />
        </div>
      </div>
      <button
        class="sc-brKeYL eScEZD"
        data-testid="header-logout-button"
      >
        Log out
      </button>
    </div>
  </div>
</div>
`;
